From mario.limonciello@amd.com  Thu Aug 31 14:29:42 2023
From: Mario Limonciello <mario.limonciello@amd.com>
Date: Thu, 31 Aug 2023 06:34:21 -0500
Subject: thunderbolt: Fix a backport error for display flickering issue
To: <stable@vger.kernel.org>
Cc: Mario Limonciello <mario.limonciello@amd.com>, <Joakim.Tjernlund@infinera.com>
Message-ID: <20230831113421.158244-1-mario.limonciello@amd.com>

From: Mario Limonciello <mario.limonciello@amd.com>

A mistake was made when backporting commit 583893a66d73 ("thunderbolt: Fix
Thunderbolt 3 display flickering issue on 2nd hot plug onwards") in missing
the `if` block.  Add it back in.

Reported-by: Joakim.Tjernlund@infinera.com
Closes: https://lore.kernel.org/stable/28b5d0accce90bedf2f75d65290c5a1302225f0f.camel@infinera.com/
Fixes: 06614ca4f18e ("thunderbolt: Fix Thunderbolt 3 display flickering issue on 2nd hot plug onwards")
Signed-off-by: Mario Limonciello <mario.limonciello@amd.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/thunderbolt/tmu.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/thunderbolt/tmu.c
+++ b/drivers/thunderbolt/tmu.c
@@ -416,6 +416,7 @@ int tb_switch_tmu_disable(struct tb_swit
 		 * mode.
 		 */
 		ret = tb_switch_tmu_rate_write(sw, TB_SWITCH_TMU_RATE_OFF);
+		if (ret)
 			return ret;
 
 		tb_port_tmu_time_sync_disable(up);
